crates/rooch/src/commands/move_cli/commands/publish.rs

```diff
@@ -131,29 +126,20 @@ impl CommandAction<ExecuteTransactionResponseView> for Publish {
         let sender = context.resolve_address(self.tx_options.sender)?.into();
         // Prepare and execute the transaction based on the action type
         let tx_result = if !self.by_move_action {
-            let (entrypoint, args) = if self.legacy {
-                let args = bcs::to_bytes(&bundles).unwrap();
-                println!("[Warning] You are publishing modules with old abi: publish_modules_entry. This is deprecated and will be removed in the future.");
-                ("publish_modules_entry".to_owned(), args)
-            } else {
-                let pkg_data = PackageData::new(
-                    MoveString::from(package.compiled_package_info.package_name.as_str()),
-                    pkg_address,
-                    bundles,
-                );
-                let pkg_bytes = bcs::to_bytes(&pkg_data).unwrap();
-                (
-                    "publish_package_entry".to_owned(),
-                    bcs::to_bytes(&pkg_bytes).unwrap(),
-                )
-            };
+            let pkg_data = PackageData::new(
+                MoveString::from(package.compiled_package_info.package_name.as_str()),
+                pkg_address,
+                bundles,
+            );
```

Comment:
In the future, we can call `build` in publish and directly use the PackageData blob from build dir.